### PR TITLE
Messaging: Added IConnectionFactory for correctly implementing disposable pattern

### PR DESCRIPTION
**Pull request by BalassaMarton**
Added IConnectionFactory for correctly implementing disposable pattern

### CODE REVIEW

File: src/messaging/dotnet/src/Client/Client/MessageRouterClient.cs [hunk 1]
```diff
@@ -185,7 +185,7 @@ private Topic GetTopic(string topicName)
     }
 
     private string? _clientId;
-    private readonly IConnection _connection;
+    private IConnection _connection;
```
Comment (ztanczos) [inline src/messaging/dotnet/src/Client/Client/MessageRouterClient.cs:188]: Nitpick: why did you remove `readonly`?

**Review by ztanczos — DISMISSED** [review]

File: src/messaging/dotnet/src/Client/Client/MessageRouterClient.cs (hunk 1 above)
Comment (BalassaMarton) [reply src/messaging/dotnet/src/Client/Client/MessageRouterClient.cs:188]: That was totally unintentional, thanks. I'll fix it in a sec.

**Review by ztanczos — APPROVED** [review]
